core/src/main/java/org/apache/brooklyn/entity/stock/AsyncApplicationImpl.java

```diff
@@ -0,0 +1,473 @@
+                throw Exceptions.propagate(e);
+            } catch (Exception e) {
+                // should remember problems, apart from those that happened starting children
+                // fixed bug introduced by the fix in dacf18b831e1e5e1383d662a873643a3c3cabac6
+                // where failures in special code at application root don't cause app to go on fire 
+                ServiceStateLogic.ServiceNotUpLogic.updateNotUpIndicator(this, START.getName(), Exceptions.collapseText(e));
+                throw Exceptions.propagate(e);
+            }
+            
+        } catch (Exception e) {
+            recordApplicationEvent(Lifecycle.ON_FIRE);
+            ServiceStateLogic.setExpectedStateRunningWithErrors(this);
+            
+            // no need to log here; the effector invocation should do that
+            throw Exceptions.propagate(e);
+            
+        } finally {
+            ServiceStateLogic.ServiceNotUpLogic.clearNotUpIndicator(this, Attributes.SERVICE_STATE_ACTUAL);
+        }
+        
+        // CHANGE FROM SUPER: NOT CALLING THESE
+        // ServiceStateLogic.setExpectedState(this, Lifecycle.RUNNING);
+        // setExpectedStateAndRecordLifecycleEvent(Lifecycle.RUNNING);
+        //
+        // logApplicationLifecycle("Started");
+    }
+
+    /**
+     * Calculates the "service.state" and "service.isUp", based on the state of the children. It 
+     * also transitions from expected=starting to expected=running once all of the children have
+     * finished their starting.
+     * 
+     * This does <em>not</em> just rely on the "service.problems" and "service.notUp.indicators"
+     * because those make different assumptions about the expected state. Instead it seems much
+     * easier to implement the specific logic for async startup here.
+     * 
+     * The important part of the implementation is {@link #onUpdated()}, and its helper methods
+     * for {@link #computeServiceUp(Lifecycle)}, {@link #computeServiceState(Lifecycle)} and
+     * {@link #computeExpectedState(Lifecycle, Lifecycle)}.
+     * 
+     * This class is not to be instantiated directly. Instead, if cusotmization is desired then override 
+     * {@link AsyncApplicationImpl#initEnrichers()} to create and add this enricher (with the same unique 
+     * tag, to replace the default).
+     */
+    public static class ServiceStateComputer extends AbstractMultipleSensorAggregator<Void> implements SensorEventListener<Object> {
+        /** standard unique tag identifying instances of this enricher at runtime */
+        public final static String DEFAULT_UNIQUE_TAG = "async-service-state-computer";
+
+        public static final ConfigKey<QuorumCheck> RUNNING_QUORUM_CHECK = ConfigKeys.builder(QuorumCheck.class, "runningQuorumCheck")
+//              .description("Logic for checking whether this service is healthy, based on children and/or members running, defaulting to requiring none to be ON-FIRE")
```

Comment:
oops, good spot - will fix those! That's the previous copy-pasted description. I meant to update those comments!